File: AutomaticDrawing.Core/Core/Painter/IPainter.vb
''' <summary>
''' 绘制器接口
''' </summary>
Public Interface IPainter
    ''' <summary>
    ''' 绘制时发生的事件
    ''' </summary>
    Event UpdatePaint(sender As Object, e As UpdatePaintEventArgs)
    ''' <summary>
    ''' 开始
    ''' </summary>
    Sub Start(lines As List(Of ILine))
    ''' <summary>
    ''' 暂停
    ''' </summary>
    Sub Pause()
    ''' <summary>
    ''' 结束
    ''' </summary>
    Sub [Stop]()
End Interface

File: AutoPaint.Core/Painter/IPainter.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AutoPaint.Core
{
    public interface IPainter
    {
        /// <summary>
        /// Gets the event that occur when a painting happens to update.
        /// </summary>
        event EventHandler<OnPaintingUpdatedEventArgs> OnPaintingUpdated;

        void Start(IPainting painting);
        void Pause();
        void Resume();
        void Stop();
    }
}
